launchdarkly/resource_launchdarkly_release_policy_order.go

```diff
@@ -0,0 +1,153 @@
+
+	// Convert interface slice to string slice
+	releasePolicyKeys := make([]string, len(policyKeys))
+	for i, key := range policyKeys {
+		releasePolicyKeys[i] = key.(string)
+	}
+
+	err := updateReleasePolicyOrder(client, projectKey, releasePolicyKeys)
+	if err != nil {
+		return diag.Errorf("failed to create release policy order for project %q: %s", projectKey, handleLdapiErr(err))
+	}
+
+	d.SetId(releasePolicyOrderID(projectKey))
+
+	return resourceReleasePolicyOrderRead(ctx, d, metaRaw)
+}
+
+func resourceReleasePolicyOrderRead(ctx context.Context, d *schema.ResourceData, metaRaw interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	client := metaRaw.(*Client)
+
+	projectKey := d.Get(PROJECT_KEY).(string)
+
+	order, err := getReleasePolicyOrder(client, projectKey)
+	if err != nil {
+		return diag.Errorf("failed to get release policy order for project %q: %s", projectKey, handleLdapiErr(err))
+	}
+
+	_ = d.Set(PROJECT_KEY, projectKey)
+	_ = d.Set(RELEASE_POLICY_KEYS, order)
+
+	return diags
+}
+
+func resourceReleasePolicyOrderUpdate(ctx context.Context, d *schema.ResourceData, metaRaw interface{}) diag.Diagnostics {
+	client := metaRaw.(*Client)
+	projectKey := d.Get(PROJECT_KEY).(string)
+
+	// If the project key changes, treat it like a new resource
+	if d.HasChange(PROJECT_KEY) {
+		return resourceReleasePolicyOrderCreate(ctx, d, metaRaw)
+	}
+
+	if d.HasChange(RELEASE_POLICY_KEYS) {
+		policyKeys := d.Get(RELEASE_POLICY_KEYS).([]interface{})
+
+		// Convert interface slice to string slice
+		releasePolicyKeys := make([]string, len(policyKeys))
+		for i, key := range policyKeys {
+			releasePolicyKeys[i] = key.(string)
+		}
+
+		err := updateReleasePolicyOrder(client, projectKey, releasePolicyKeys)
+		if err != nil {
+			return diag.Errorf("failed to update release policy order for project %q: %s", projectKey, handleLdapiErr(err))
+		}
+	}
+
+	return resourceReleasePolicyOrderRead(ctx, d, metaRaw)
+}
+
+func resourceReleasePolicyOrderDelete(ctx context.Context, d *schema.ResourceData, metaRaw interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	// Just remove from state, don't actually delete anything
+	d.SetId("")
+	return diags
+}
+
+func resourceReleasePolicyOrderExists(d *schema.ResourceData, metaRaw interface{}) (bool, error) {
+	client := metaRaw.(*Client)
+	projectKey := d.Get(PROJECT_KEY).(string)
+
+	_, err := getReleasePolicyOrder(client, projectKey)
+	if err != nil {
+		return false, fmt.Errorf("failed to get release policy order for project %q: %s", projectKey, handleLdapiErr(err))
+	}
+
+	return true, nil
+}
+
+func resourceReleasePolicyOrderImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	importID := d.Id()
+	if importID == "" {
+		return nil, fmt.Errorf("import ID cannot be empty")
+	}
+
+	// Parse the import ID to extract the project key
+	projectKey := strings.TrimSuffix(importID, "/release-policy-order")
```

Comment:
Do we need the suffix here? I wouldn't expect us to need to since this resource is uniquely identified by project key.

Reply:
😅 TIL! I'll update it! Thanks Drew